使用了外部依赖的插件在开发模式下出现 NoClassDefFoundError 问题

### What is version of Halo has the issue?

latest

### What database are you using?

H2

### What is your deployment method?

Docker

### Your site address.

_No response_

### What happened?

此问题仅开发模式存在，导致无法使用开发模式，jar 包可以正常使用。

### Relevant log output
异常示例：
```shell
Caused by: java. lang. NoClassDefFoundEenon: Create breakpoint : com/wechat/pay/java/core/Config
at java. base/java.Lang. Class. getDeclaredMethods0 (Native Method) -[na:na]
at java. base/java. Lang. Class.privateGetDeclaredMethods (Class, java: 3402) -[na:na]
at java. base/java. lang. Class. getDeclaredMethods(Class. java:2504) ~ [na:na]
at
org. springframework.utit.ReflectionUtils.getDeclaredMethods(Reflectionlitils.java:465)-[spring-core-6.0.3.jar:6.0.3]
32 common frames omitted
Caused by: java. Lang. ClassNotFoundExcention Create breakpoint com.wechat.pay-java.core.Config
```
```
Caused by: java.lang.NoClassDefFoundError: com/aliyun/oss/OSSException
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method) ~[na:na]
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3402) ~[na:na]
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2504) ~[na:na]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:465) ~[spring-core-6.0.3.jar:6.0.3]
	... 33 common frames omitted
Caused by: java.lang.ClassNotFoundException: com.aliyun.oss.OSSException
	at org.pf4j.PluginClassLoader.loadClass(PluginClassLoader.java:144) ~[pf4j-3.8.0.jar:3.8.0]
	... 37 common frames omitted
```

### Additional information

/kind improvement
/milestone 2.2.x
/assign

--- a/src/main/java/run/halo/app/plugin/PluginAutoConfiguration.java
+++ b/src/main/java/run/halo/app/plugin/PluginAutoConfiguration.java
@@ -107,17 +107,30 @@ public class PluginAutoConfiguration {
                     } else {
                         return new CompoundPluginLoader()
                             .add(new DevelopmentPluginLoader(this) {
+
+                                @Override
+                                protected PluginClassLoader createPluginClassLoader(Path pluginPath,
+                                    PluginDescriptor pluginDescriptor) {
+                                    return new PluginClassLoader(pluginManager, pluginDescriptor,
+                                        getClass().getClassLoader(), ClassLoadingStrategy.APD);
+                                }
+
                                 @Override
                                 public ClassLoader loadPlugin(Path pluginPath,
                                     PluginDescriptor pluginDescriptor) {
-                                    PluginClassLoader pluginClassLoader =
-                                        new PluginClassLoader(pluginManager, pluginDescriptor,
-                                            getClass().getClassLoader(), ClassLoadingStrategy.APD);
-
-                                    loadClasses(pluginPath, pluginClassLoader);
-                                    loadJars(pluginPath, pluginClassLoader);
-
-                                    return pluginClassLoader;
+                                    if (pluginProperties.getClassesDirectories() != null) {
+                                        for (String classesDirectory :
+                                            pluginProperties.getClassesDirectories()) {
+                                            pluginClasspath.addClassesDirectories(classesDirectory);
+                                        }
+                                    }
+                                    if (pluginProperties.getLibDirectories() != null) {
+                                        for (String libDirectory :
+                                            pluginProperties.getLibDirectories()) {
+                                            pluginClasspath.addJarsDirectories(libDirectory);
+                                        }
+                                    }
+                                    return super.loadPlugin(pluginPath, pluginDescriptor);
                                 }
                             }, this::isDevelopment)
                             .add(new JarPluginLoader(this) {

--- a/src/main/java/run/halo/app/plugin/PluginProperties.java
+++ b/src/main/java/run/halo/app/plugin/PluginProperties.java
@@ -17,6 +17,7 @@ import org.springframework.boot.context.properties.ConfigurationProperties;
 @Data
 @ConfigurationProperties(prefix = "halo.plugin")
 public class PluginProperties {
+    public static final String GRADLE_LIBS_DIR = "build/libs";
 
     /**
      * Auto start plugin when main app is ready.
@@ -53,7 +54,7 @@ public class PluginProperties {
     /**
      * Extended Plugin Jar Directory.
      */
-    private List<String> libDirectories = new ArrayList<>();
+    private List<String> libDirectories = new ArrayList<>(List.of(GRADLE_LIBS_DIR));
 
     /**
      * Runtime Mode：development/deployment.

--- a/src/main/resources/application-dev.yaml
+++ b/src/main/resources/application-dev.yaml
@@ -19,11 +19,7 @@ halo:
       super-admin-password: admin
   plugin:
     runtime-mode: development # development, deployment
-    classes-directories:
-      - "build/classes"
-      - "build/resources"
-    lib-directories:
-      - "libs"
+    plugins-root: ${halo.work-dir}/plugins
   work-dir: ${user.home}/halo2-dev
 logging:
   level:

--- a/src/main/resources/application-dev.yaml
+++ b/src/main/resources/application-dev.yaml
@@ -19,7 +19,6 @@ halo:
       super-admin-password: admin
   plugin:
     runtime-mode: development # development, deployment
-    plugins-root: ${halo.work-dir}/plugins
   work-dir: ${user.home}/halo2-dev
 logging:
   level:

